src/tasks/__tests__/processBulkUpload.int.test.ts

```diff
@@ -231,6 +270,24 @@ describe('processBulkUpload', () => {
     expect(updatedBulkUpload.status).toEqual(BulkUploadStatus.FAILED);
   });
 
+  it('should move the csv file to processed location if the csv contains an invalid short code', async () => {
+    await createTestBaseFields();
+    const sourceKey = '550e8400-e29b-41d4-a716-446655440000';
```

Comment:
Should this also start with `unprocessed/`? In fact, should we just extract a `TEST_SOURCE_KEY` constant?

Reply:
Yeah, I was thinking we should probably add a status pre-check to the worker to make sure (1) the state of the bulk upload is `PENDING` and (2) the key has the `unprocesed` prefix.

Once implemented, these tests would need to be updated (currently, of course, that is not a precondition which is why the tests don't fail)

Reply:
I made this change and also added those checks (without the checks it would be too easy for a malicious user to predict a processed file's key path, allowing them to process someone else's bulk upload as though it was their own *AND* resulting in that processed file being moved to a new location).

Reply:
Oh, woah, fantastic catch - I hadn't even thought about that kind of attack! I was just thinking the tests should match the kind of data we expect in production. Great work, @slifty!